### PR TITLE
Believe the quoted weights in the transit fit's uncertainties

### LINKED ISSUE #699: Transit fit still rescales uncertainties by redchi (scale_covar default) — inconsistent with #696
#696 turned off lmfit's default `scale_covar` rescaling in `transform_to_catalog`, so the transform fit now reports coefficient and magnitude uncertainties measured against the errors as quoted, with `fit_redchi` / `fit_excess_scatter` carrying the goodness-of-fit information separately (#690).

The transit fit still uses the old convention. `TransitModelFit._run_fit` (`stellarphot/transit_fitting/core.py`) weights the residual by `1 / flux_errors` and calls `lmfit.minimize` without `scale_covar=False`, so lmfit's default rescaling is on: transit depth/duration/center uncertainties are silently multiplied by the square root of the reduced chi-square — the behavior #690 called misleading for the magnitude transform.

The repo's two weighted fits now report uncertainties under contradictory conventions. Whichever convention is right, it should be one shared decision:

- switch the transit fit to `scale_covar=False` and expose the reduced chi-square (and possibly an excess-scatter estimate) the way `transform_to_catalog` now does, or
- document explicitly that the transit fit intentionally keeps rescaled uncertainties and why.

Neither #697 nor #698 covers this.

*This issue was written by Claude Code at Matt's direction.*

### PR DESCRIPTION
**Pull request by mwcraig**
Closes #699 — brings `TransitModelFit` onto the uncertainty convention #696 adopted for `transform_to_catalog`.

- `_run_fit` passes `scale_covar=self.weights is None`: a weighted fit's `stderr` values now believe the quoted flux errors; an unweighted fit keeps the rescaling since the residuals are its only noise estimate. Before this, errors quoted 10× too small gave the *same* `stderr` (the failing test showed 0.01133 vs 0.01135).
- New attributes `fit_redchi` and `fit_excess_scatter` (None before `fit()`, NaN excess when unweighted), computed before any state is assigned so a failing fit still leaves the instance untouched.
- `_quoted_redchi`/`_excess_scatter` move from `magnitude_transforms.py` to a new `stellarphot.utils.fit_diagnostics` module as public `quoted_redchi`/`excess_scatter` (numpy + `scipy.optimize.brentq` only; automodapi entry added; docs build with `-W` passes). No behaviour change on the transform side.
- `compare_detrend_options` unchanged (BIC is independent of `scale_covar`). No weight floor in this PR.
- Tests: stderr ratio 0.1 between under-quoted and truthful weights with `fit_redchi ≈ 100`; excess ≈ `sqrt(noise² − quoted²)`; unweighted stderr = truthful stderr × `sqrt(fit_redchi)` (the exact relation `scale_covar` implements); attributes None before fit.
- One markdown cell added after `mod.params` in both TESS model-fit notebooks; CHANGES entry.

This PR was written by Claude Code at Matt's direction.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_015RURMHNvNLcGuqnx4u7D4V

### CODE REVIEW

File: stellarphot/transit_fitting/core.py [hunk 1]
```diff
@@ -500,7 +544,18 @@ def fit(self):
 
         result = self._run_fit(self.params)
 
+        # Every diagnostic is computed before any state is assigned, so a
+        # fit that raises leaves the instance exactly as it was.
+        if self.weights is None:
+            sigma = None
+        else:
+            sigma = 1 / np.asarray(self.weights, dtype=float)
```
Comment (mwcraig) [inline stellarphot/transit_fitting/core.py]: **Regression: a zero weight now makes `fit()` raise after lmfit succeeded.** `weights` is an unvalidated attribute and lmfit tolerates `weight == 0` (the point is just ignored), which is the natural way to mask a point. On `main` that fit succeeds. Here `sigma` becomes `inf`, `residual / weights` in `excess_scatter` is `0/0 = NaN`, and when the quoted reduced chi-square is above 1 the bracket function evaluates NaN at zero. Reproduced on this branch with `weights[3] = 0` and under-quoted errors:

```
ValueError: The function value at x=0.0 is NaN; solver cannot continue.
```

plus a `divide by zero` RuntimeWarning here (an error under the repo's `filterwarnings = error`). When redchi ≤ 1 it silently returns `excess = 0`, so the failure is data-dependent. Suggest masking `weights == 0` points out of the diagnostics (they contribute nothing to chi-square anyway), or validating weights and documenting the precondition. Copilot's `0/0` comment is this same issue; it's a crash, not a NaN result.

*Review written by Claude Code at Matt's direction.*

File: stellarphot/transit_fitting/core.py [hunk 2]
```diff
@@ -500,7 +544,18 @@ def fit(self):
...
+        redchi = result.redchi
```
Comment (mwcraig) [inline stellarphot/transit_fitting/core.py]: Use `quoted_redchi(result, sigma, self.weights)` here rather than `result.redchi`. lmfit sums `residual**2` while the bracket in `excess_scatter` sums `(residual/w)**2 / sigma**2`, and the two round differently: in a 200k-trial simulation of fits whose reduced chi-square sits within a few ULP of 1, ~3% had lmfit's value `> 1` while the bracket value at zero was `≤ 1`, and `brentq` then raises on two negative endpoints (scipy only tolerates `f(a) == 0` exactly). Practically this needs redchi within ~1e-15 of 1, so the risk is tiny — but the whole reason `redchi_quoted` is passed in (per the comments in `fit_diagnostics.py`) is that the gate value *is* the reported value, and this caller breaks that contract for free. `quoted_redchi` returns `result.redchi` unchanged when `sigma is None`, so the unweighted test still holds.

File: stellarphot/utils/fit_diagnostics.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    if sigma is None:
+        return fit_result.redchi
+
+    residual = np.asarray(fit_result.residual) / weights
+    # The same operand order as `excess_scatter`'s bracket function at zero
+    # excess -- ``residual**2 / sigma**2``, not ``(residual / sigma)**2``.
+    # The gate there compares this value to one, and a value that disagreed
+    # with the bracket by one ULP around it could hand
+    # `~scipy.optimize.brentq` two negative endpoints -- a `ValueError`.
+    return float(np.sum(residual**2 / sigma**2) / fit_result.nfree)
```
Comment (mwcraig) [inline stellarphot/utils/fit_diagnostics.py]: Now that this is public: lmfit computes `redchi = chisqr / max(1, nfree)`, while this divides by `nfree` unguarded, so `nfree == 0` gives `inf`/`ZeroDivisionError` where lmfit gives `chisqr`. `excess_scatter` guards `nfree <= 0` but this doesn't. Pre-existing from #696, but worth either mirroring lmfit or stating the precondition — and if the transit caller switches to `quoted_redchi` (comment above) an over-parameterized transit fit becomes exposed to it.

File: stellarphot/utils/fit_diagnostics.py [hunk 4]
```diff
@@ -0,0 +1,176 @@
+"""
+Diagnostics of a weighted least-squares fit: how the residuals compare to the
+uncertainties the points were quoted with.
+
+Shared by `~stellarphot.utils.magnitude_transforms.transform_to_catalog` and
+`~stellarphot.transit_fitting.TransitModelFit`, both of which fit with
+``scale_covar=False`` (issues #690 and #699) and report these numbers instead
```
Comment (mwcraig) [inline stellarphot/utils/fit_diagnostics.py]: Not literally true: both callers use `scale_covar=(weights/sigma) is None` (`magnitude_transforms.py:1400`, `core.py:374`), so unweighted fits keep the rescaling. "both of which turn off the rescaling for a weighted fit" would match. (Agree with Copilot here.)

**Review by mwcraig — COMMENTED** [review]
Critical review of the branch at 16daf58. Ran the transit and magnitude-transform test files locally (160 passed, 2 remote-data skipped); the `_excess_scatter` → `excess_scatter` rename is complete, there is no circular import, the new automodapi entry is needed (the functions aren't in `magnitude_transforms.__all__`), the unweighted stderr × √redchi relation the test asserts is exactly what lmfit implements, and the diagnostics really are computed before any state is assigned. Inline comments cover the substantive points; the one that should block is the zero-weight regression on `core.py` line 552.

Three smaller things not worth an inline thread:

- `compare_detrend_options()` calls `_run_fit` per combination and never `fit()`, so after `compare_detrend_options()` (default `apply_best=False`) the new `fit_redchi` / `fit_excess_scatter` still describe whatever fit came *before* the comparison. That's consistent with how `fit_result` already behaves, but the new attributes invite "check the diagnostics after comparing", so a sentence in their attribute docs would head that off.

- `test_fit_stderr_believes_quoted_weights` takes ~7.8 s (the under-quoted fit converges slowly); the other three new tests take 0.03 s combined. Not blocking, but it's now the slowest test in the file.
- CHANGES: "follows the same convention" implicitly means the #690 entry three bullets up; "the convention #690 adopted for `transform_to_catalog`" would make that explicit.

*This review was written by Claude Code at Matt's direction.*

File: stellarphot/utils/fit_diagnostics.py [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+    redchi_quoted : float, optional
+        ``quoted_redchi(fit_result, sigma, weights)`` for this fit, if the
+        caller already has it -- computed here otherwise. Ignored when
+        ``sigma`` is `None`.
+
+    Returns
+    -------
+    float
+        The excess scatter in magnitudes; zero when the residuals are already
```
Comment (mwcraig) [inline stellarphot/utils/fit_diagnostics.py]: "in magnitudes" is wrong for the transit caller, where this is in flux units — "in the units of `sigma`" covers both. The flat-field / catalog-photometry sentence in the summary paragraph is also transform-specific; fine to keep if labelled as that case. (Agree with Copilot here.)

File: stellarphot/transit_fitting/core.py [hunk 6]
```diff
@@ -500,7 +544,18 @@ def fit(self):
...
+        excess = excess_scatter(result, sigma, self.weights, redchi_quoted=redchi)
```
[the remark this answers is not in the document]
Comment (mwcraig) [reply stellarphot/transit_fitting/core.py]: Agreed — and this is the same point as the review comment on line 553. The helper's whole reason for taking `redchi_quoted` is that the gate is the reported float, and `_run_fit`'s caller dropped that by handing over `result.redchi`. Will compute `redchi = quoted_redchi(result, sigma, self.weights)` and pass that through; `quoted_redchi` returns `result.redchi` unchanged when `sigma is None`, so the unweighted path and its test are unaffected.

— _Written by Claude at @mwcraig's direction._

File: stellarphot/utils/fit_diagnostics.py [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+
+    This is also the value `excess_scatter` gates on: whether any scatter
+    needs inventing at all is exactly the question of whether this reduced
+    chi-square already sits at or below one, so the two share this one
+    computation rather than each summing the same quantity independently.
+    """
+    if sigma is None:
+        return fit_result.redchi
+
+    residual = np.asarray(fit_result.residual) / weights
```
[the remark this answers is not in the document]
Comment (mwcraig) [reply stellarphot/utils/fit_diagnostics.py]: Confirmed — reproduced with a 30-point linear fit, `weights[3] = 0`: lmfit fits fine, then both `quoted_redchi` and `excess_scatter` hit `0 / 0` here (a `RuntimeWarning`, which the repo's `filterwarnings = error` turns into an exception under pytest). Will fix in both places by dropping zero-weight points from the sums rather than documenting a contract: they contribute exactly zero to chi-square, so the result for every other input is unchanged. Concretely, `good = np.broadcast_to(weights, residual.shape) != 0` and then index `residual`, `weights` and `sigma` by `good`; `nfree` stays lmfit's, which counts those points in `ndata` — matching the unweighted branch that returns `fit_result.redchi` as-is.

— _Written by Claude at @mwcraig's direction._

File: stellarphot/utils/fit_diagnostics.py [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+        ``quoted_redchi(fit_result, sigma, weights)`` for this fit, if the
+        caller already has it -- computed here otherwise. Ignored when
+        ``sigma`` is `None`.
+
+    Returns
+    -------
+    float
+        The excess scatter in magnitudes; zero when the residuals are already
+        no larger than the errors claim, and NaN for an unweighted fit, whose
+        residuals have no errors to be excessive with respect to.
```
[the remark this answers is not in the document]
Comment (mwcraig) [reply stellarphot/utils/fit_diagnostics.py]: Agreed — "in magnitudes" was carried over from when this lived in `magnitude_transforms.py`. Will change to "in the units of `sigma`" and label the flat-field / catalog sentence in the summary as the transform case.

— _Written by Claude at @mwcraig's direction._

File: stellarphot/utils/fit_diagnostics.py [hunk 9]
```diff
@@ -0,0 +1,176 @@
+"""
+Diagnostics of a weighted least-squares fit: how the residuals compare to the
+uncertainties the points were quoted with.
+
+Shared by `~stellarphot.utils.magnitude_transforms.transform_to_catalog` and
+`~stellarphot.transit_fitting.TransitModelFit`, both of which fit with
+``scale_covar=False`` (issues #690 and #699) and report these numbers instead
+of letting `lmfit` rescale the covariance to make the reduced chi-square one.
```
[the remark this answers is not in the document]
Comment (mwcraig) [reply stellarphot/utils/fit_diagnostics.py]: Agreed — both callers use `scale_covar=(sigma is None)` (`magnitude_transforms.py:1400`, `core.py:374`), so only the weighted fit turns the rescaling off. Will reword to "both of which turn off the rescaling for a weighted fit".

— _Written by Claude at @mwcraig's direction._

File: stellarphot/transit_fitting/core.py (hunk 1 above)
Comment (mwcraig) [reply stellarphot/transit_fitting/core.py]: Confirmed, and it's worse than data-dependent under the test suite: the `divide by zero` `RuntimeWarning` fires in `residual / weights` before the redchi gate, so with `filterwarnings = error` any zero weight raises regardless of redchi. Outside pytest it's as described — silent `excess = 0` when redchi ≤ 1, `ValueError` from `brentq` otherwise. Copilot's suppressed comment on `fit_diagnostics.py:153` and its inline one on line 64 are this same bug.

Going with masking rather than validation, since zero weights are lmfit's own masking idiom and `main` accepts them: in both `quoted_redchi` and `excess_scatter`, select `weights != 0` (broadcast, so a scalar weight still works) before dividing. Those points contribute zero to chi-square so nothing else changes; `nfree` stays lmfit's value. Will add a test with a zero weight under both redchi > 1 and ≤ 1, plus a `quoted_redchi` test.

— _Written by Claude at @mwcraig's direction._

File: stellarphot/transit_fitting/core.py (hunk 2 above)
Comment (mwcraig) [reply stellarphot/transit_fitting/core.py]: Agreed; same fix as Copilot's comment on line 554 — `redchi = quoted_redchi(result, sigma, self.weights)` and pass it as `redchi_quoted`. This also makes the transit caller consistent with `transform_to_catalog`, which already does exactly that.

— _Written by Claude at @mwcraig's direction._

File: stellarphot/utils/fit_diagnostics.py (hunk 3 above)
Comment (mwcraig) [reply stellarphot/utils/fit_diagnostics.py]: Verified against lmfit 1.3.4: `MinimizerResult._calculate_statistics` does `self.redchi = self.chisqr / max(1, self.nfree)`. Will mirror that here (`/ max(1, fit_result.nfree)`) so `quoted_redchi` agrees with `fit_result.redchi` in the degenerate case too, which matters now that the transit caller will route through it. `excess_scatter`'s `nfree <= 0` → NaN stays as is; that one is a separate question (there is no scatter to find with no degrees of freedom).

— _Written by Claude at @mwcraig's direction._

File: stellarphot/utils/fit_diagnostics.py (hunk 4 above)
Comment (mwcraig) [reply stellarphot/utils/fit_diagnostics.py]: Agreed — will use that wording. Same as Copilot's comment on line 8.

— _Written by Claude at @mwcraig's direction._

File: stellarphot/utils/fit_diagnostics.py (hunk 5 above)
Comment (mwcraig) [reply stellarphot/utils/fit_diagnostics.py]: Agreed — "in the units of `sigma`", and the flat-field / catalog sentence gets labelled as the magnitude-transform case. Same as Copilot's comment on line 110.

— _Written by Claude at @mwcraig's direction._

File: stellarphot/utils/fit_diagnostics.py [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+        The excess scatter, in the units of ``sigma``; zero when the
+        residuals are already no larger than the errors claim, and NaN for
+        an unweighted fit, whose residuals have no errors to be excessive
+        with respect to.
+
+    Notes
+    -----
+    Reported rather than folded into the weights. A fit that absorbs its own
+    excess scatter has a reduced chi-square of one by construction, which
+    destroys the one diagnostic that revealed any of this.
```
Comment (mwcraig) [inline stellarphot/utils/fit_diagnostics.py:122]: It is completely unclear what "this" refers to, whichmeans it probably isn't needed

File: stellarphot/utils/fit_diagnostics.py [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+    that is the same for every point barely changes where a fit lands -- it
+    rescales the weights nearly uniformly -- and holding them fixed keeps this
+    a description of the fit that was actually reported.
+
+    Callers such as
+    `~stellarphot.utils.magnitude_transforms.transform_to_catalog` pass
+    ``redchi_quoted`` in so that the value gating this function is the
+    exact float they report as ``fit_redchi``,
+    rather than a second sum of the same quantity that could round
+    differently by a few ULPs around 1.0.
```
Comment (mwcraig) [inline stellarphot/utils/fit_diagnostics.py]: Do we really need to worry about rounding at machine precision???

File: stellarphot/utils/fit_diagnostics.py (hunk 10 above)
Comment (mwcraig) [reply stellarphot/utils/fit_diagnostics.py:122]: Gone in 7b9fada. The paragraph existed to justify the `redchi_quoted` parameter, and that parameter is gone too: `excess_scatter` now gates on its own bracket function evaluated at zero, so there is only one sum and nothing for "this" to refer to.

— _Written by Claude at @mwcraig's direction._

File: stellarphot/utils/fit_diagnostics.py (hunk 11 above)
Comment (mwcraig) [reply stellarphot/utils/fit_diagnostics.py]: No — and in 7b9fada we no longer do. Both ULP guards were defending against computing the same number twice. The gate now reuses the bracket function (`f(0) <= 0` → return 0), and the upper endpoint is `2 * rms` instead of `rms`, which makes `f(upper) <= -3/4` by construction rather than "at most zero, barring rounding". The rounds-to-positive guard and its comment are deleted; the existing regression tests for both cases still pass.

— _Written by Claude at @mwcraig's direction._
